### PR TITLE
fixes `pyximport.install(pyimport=True)` on the darwin architecture

### PR DESCRIPTION
**Pull request by megawidget**
On OS X, `distutils` without `sysconfig` appears to shadow `distutils.sysconfig`.  Specifically, `distutils/utils.py` tries to import `distutils.sysconfig` and fails.

### CODE REVIEW

**Comment by megawidget** [comment]
Yikes, what's happening with the tests? O.o

**Comment by scoder** [comment]
The test log shows this:
```
/home/travis/virtualenv/python2.7.14/bin/python -c "import pyimport_test; pyimport_test.test()"

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "pyimport_test.py", line 13, in test
    assert not mymodule.__file__.rstrip('oc').endswith('.py'), mymodule.__file__
AssertionError: mymodule.py

FAIL
```
Also, the test runs for something ~45 seconds, which is really long.

Maybe `pyximport(pyimport)` could benefit from a pattern option that allows to restrict the modules it attempts to compile…

File: pyximport/pyximport.py [hunk 1]
```diff
@@ -351,6 +351,9 @@ def __init__(self, pyxbuild_dir=None, inplace=False, language_level=None):
         self.uncompilable_modules = {}
         self.blocked_modules = ['Cython', 'pyxbuild', 'pyximport.pyxbuild',
                                 'distutils.extension', 'distutils.sysconfig']
+        # on OS X, distutils appears to shadow distutils.sysconfig
+        if os.name is 'posix' and os.uname()[0].lower().startswith('darwin'):
+            self.blocked_modules.append('distutils')
```
Comment (scoder) [inline pyximport/pyximport.py]: I think it's reasonable to _always_ block all of distutils. Compiling the compiling infrastructure while it's being used just unnecessarily screams for trouble.

Comment (megawidget) [reply pyximport/pyximport.py]: good point; i wanted to be as surgical as possible, but this is cleaner and better

**Comment by megawidget** [comment]
> Maybe `pyximport(pyimport)` could benefit from a pattern option that allows to restrict the modules it attempts to compile…

I would be happy to add that in a separate PR.  Meanwhile, something needs to be done about the tests and I'm rather at a loss -- I assume they are failing for `master` as well?
Maybe I could take a gander at them over the weekend if nobody else gets to them first.

**Comment by scoder** [comment]
I think the test failures might be a corrupted `ccache` on travis. I'll see if I can fix it.

**Comment by scoder** [comment]
No, found the problem. We were calling mypy incorrectly in one of the integration tests and the latest mypy release failed with an error. Fixed in master now.

**Pull request merged**